scripts/make_gitrev.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+file=include/spl/sys/zfs_gitrev.h
+
+#
+# Set default file contents in case we bail.
+#
+rm -f $file
+echo "#define\tZFS_META_GITREV \"unknown\"" >>$file
+
+#
+# Check if we are in a git repo.
+#
+git rev-parse --git-dir > /dev/null 2>&1 || exit
+
+#
+# Check if there are uncommitted changes
+#
+git diff-index --quiet HEAD -- || exit
```

Comment:
Should we perhaps still generate a hash with "(modified)" appended to it, just so that people aren't too confused why it's suddenly "unknown"?
Also nit: `--` looks unnecessary

Reply:
For simplicity, I'm going to drop this (adding `(modified)`) if that's OK with you.

Reply:
It's up to you. That should be very easy to implement though.